node/src/types/status_feed.rs

```diff
@@ -35,9 +45,39 @@ static GET_STATUS_RESULT: Lazy<GetStatusResult> = Lazy::new(|| {
         chainspec_info: ChainspecInfo::doc_example().clone(),
         version: crate::VERSION_STRING.as_str(),
     };
-    GetStatusResult::from(status_feed)
+    GetStatusResult::new(status_feed, DOCS_EXAMPLE_PROTOCOL_VERSION.clone())
 });
 
+/// Summary information from the chainspec.
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct ChainspecInfo {
+    /// Name of the network.
+    name: String,
+    /// If `Some` then genesis or upgrade process returned a valid post state hash.
+    state_root_hash: Option<Digest>,
+    next_upgrade: Option<NextUpgrade>,
+}
+
+impl DocExample for ChainspecInfo {
+    fn doc_example() -> &'static Self {
+        &*CHAINSPEC_INFO
+    }
+}
+
+impl ChainspecInfo {
+    pub(crate) fn new(
+        chainspec_network_name: String,
+        state_root_hash: Option<Digest>,
+        next_upgrade: Option<NextUpgrade>,
+    ) -> Self {
+        ChainspecInfo {
+            name: chainspec_network_name,
+            state_root_hash,
```

Comment:
This comes from the `ChainspecLoader`, if there was an upgrade will that be a hash of the state root after applying the GlobalState changes or before them?

Reply:
That's after the upgrade.